comps/finetuning/handlers.py

```diff
@@ -190,16 +191,20 @@ def handle_list_finetuning_checkpoints(request: FineTuningJobIDRequest):
     checkpoints = []
     if os.path.exists(output_dir):
         # Iterate over the contents of the directory and add an entry for each
-        for _ in os.listdir(output_dir):  # Loop over directory contents
-            checkpointsResponse = FineTuningJobCheckpoint(
-                id=f"ftckpt-{uuid.uuid4()}",  # Generate a unique ID
-                created_at=int(time.time()),  # Use the current timestamp
-                fine_tuned_model_checkpoint=output_dir,  # Directory path itself
-                fine_tuning_job_id=fine_tuning_job_id,
-                object="fine_tuning.job.checkpoint",
-            )
-            checkpoints.append(checkpointsResponse)
-            checkpoint_id_to_checkpoint_path[checkpointsResponse.id] = checkpointsResponse.fine_tuned_model_checkpoint
+        files = os.listdir(output_dir)
+        for file in files:  # Loop over directory contents
+            file_path = os.path.join(output_dir, file)
+            if os.path.isdir(file_path) and file.startswith("checkpoint"):
```

Comment:
output_dir is the path of the last saved checkpoint, this way would miss this checkpoint.

Reply:
updated